### PR TITLE
[Datasets] Configure swagger parameters + path + security

### PR DESCRIPTION
**Pull request by youennmerel**
# Changes
- Add a keycloak client "shanoir-swagger"
- Correct @Param annotation to allow param input in datasets Swagger UI
- Add 2 authentication method (bearer, Oauth2) to allow authentication in datasets Swagger UI

# How to test
- Access datasets Swagger UI : https://shanoir-qualif.irisa.fr/shanoir-ng/datasets/swagger-ui/index.html
- Authenticate with OAuth2 method (click on "Authorize :lock:" button)
> image

(keep client_secret empty)
- Test some endpoints
- Authenticate with Bearer method (click on "Authorize :lock:" button)
Use token from browser console
- Test some endpoints

### CODE REVIEW

File: docker-compose/keycloak/cfg/shanoir-ng-realm.json [hunk 1]
```diff
@@ -1004,6 +1004,96 @@
...
+      "name": "shanoir-swagger",
+      "description": "",
+      "rootUrl": "",
+      "adminUrl": "",
+      "baseUrl": "",
+      "surrogateAuthRequired": false,
+      "enabled": true,
+      "alwaysDisplayInConsole": false,
+      "clientAuthenticatorType": "client-secret",
+      "redirectUris": [
```
Comment (jcomedouteau) [inline docker-compose/keycloak/cfg/shanoir-ng-realm.json:1018]: Use      
 "redirectUris": [
        "SHANOIR_URL_SCHEME://SHANOIR_URL_HOST/*"
      ],

?

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java [hunk 2]
```diff
@@ -140,7 +140,7 @@ public class DatasetApiController implements DatasetApi {
 
 	@Override
 	public ResponseEntity<Void> deleteDataset(
-			@Parameter(name = "id of the dataset", required = true) @PathVariable("datasetId") final Long datasetId) throws EntityNotFoundException, RestServiceException {
```
Comment (jcomedouteau) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java:143]: Is it normal ?

**Review by jcomedouteau — APPROVED** [review]
Just the little remark on [shanoir-ng-realm.json](https://github.com/fli-iam/shanoir-ng/pull/2252/files#diff-61e5f5eb2d58b953e39f422d503b87671f3c98c8face5c30435d5f6611352e7d) then we are ok :)

**Comment by youennmerel** [comment]
Hi @jcomedouteau,
I've updated `redirectUris` in [shanoir-ng-realm.json](https://github.com/fli-iam/shanoir-ng/pull/2252/files#diff-61e5f5eb2d58b953e39f422d503b87671f3c98c8face5c30435d5f6611352e7d) and tested it on qualif.

All's right

**Review by jcomedouteau — APPROVED** [review]

**Pull request merged**